bin/faucet/src/faucet/mod.rs

```diff
@@ -187,24 +190,37 @@ impl Faucet {
     pub async fn run(
         mut self,
         mut rpc_client: RpcClient,
-        mut requests: mpsc::Receiver<(MintRequest, oneshot::Sender<(BlockNumber, Note)>)>,
+        mut requests: mpsc::Receiver<(MintRequest, Sender<(BlockNumber, Note)>)>,
     ) -> anyhow::Result<()> {
         let coin_seed: [u64; 4] = random();
-        let rng = RpoRandomCoin::new(coin_seed.map(Felt::new));
-
-        while let Some((request, response_sender)) = requests.recv().await {
-            // Skip doing work if the user no longer cares about the result.
-            if response_sender.is_closed() {
-                tracing::info!(request.account_id=%request.account_id, "request cancelled");
-                continue;
-            }
+        let mut rng = RpoRandomCoin::new(coin_seed.map(Felt::new));
+
+        let mut buffer = Vec::new();
+        let limit = 100; // we could include 256 notes per tx, but requests channel is limited to 100 atm
```

Comment:
Could you post the numbers for executing and proving comparing the different amount of notes that we discussed?